libs/wingc/src/lib.rs

```diff
@@ -313,7 +313,7 @@ pub fn compile(
 	type_check(&mut scope, &mut types, &source_path, &mut jsii_types);
 
 	// Validate that every Expr in the final tree has been type checked
-	let mut tc_assert = TypeCheckAssert::new(&types);
+	let mut tc_assert = TypeCheckAssert::new(&types, found_errors());
```

Comment:
found_errors() can be true even if there are no type check errors (if there are parser errors). Does that hurt the usefulness of this extra check?

Reply:
No, I'll fix the comment above to better reflect what's being done here now. The check verifies the type-checker didn't miss anything, which in practice is:
1. All expressions were resolved to something.
2. If any were resolved to `Unresolved` then we should have generated some error diagnostics (otherwise we'll probably mess something up downstream). I need the new `found_errors()` arg for this second check.